### PR TITLE
Use string length to cover corner case for decimal number count

### PR DESCRIPTION
**Pull request by gtxistxgao**
---

There are few corner case in python that a float number time 100000 and remain 1 will result in a large number. In this PR, we convert the float to string and directly check the decimal number count.

**Python 3.8.2**

print(0.061 * 100000 % 1) => 0.0
print(0.062 * 100000 % 1) => 0.0
print(0.063 * 100000 % 1) => 0.0
print(0.064 * 100000 % 1) => 0.0
print(0.065 * 100000 % 1) => 0.0
print(0.066 * 100000 % 1) => 0.0
print(0.067 * 100000 % 1) => 0.0
print(0.068 * 100000 % 1) => 9.094947017729282e-13
print(0.069 * 100000 % 1) => 9.094947017729282e-13
print(0.070 * 100000 % 1) => 9.094947017729282e-13
print(0.071 * 100000 % 1) => 0.9999999999990905
print(0.072 * 100000 % 1) => 0.9999999999990905
print(0.073 * 100000 % 1) => 0.9999999999990905
print(0.074 * 100000 % 1) => 0.0
print(0.075 * 100000 % 1) => 0.0
print(0.076 * 100000 % 1) => 0.0

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/aks-preview/azext_aks_preview/_validators.py [hunk 1]
```diff
@@ -214,7 +214,7 @@ def validate_spot_max_price(namespace):
     if not isnan(namespace.spot_max_price):
         if namespace.priority != "Spot":
             raise CLIError("--spot_max_price can only be set when --priority is Spot")
-        if namespace.spot_max_price > 0 and not isclose(namespace.spot_max_price * 100000 % 1, 0, rel_tol=1e-06):
+        if len(str(namespace.spot_max_price).split(".")) > 1 and len(str(namespace.spot_max_price).split(".")[1]) > 5:
```
[the remark this answers is not in the document]
Comment (qwordy) [reply src/aks-preview/azext_aks_preview/_validators.py:217]: Could you add a comment for this line? Tricky code.

File: src/aks-preview/azext_aks_preview/tests/latest/test_validators.py [hunk 2]
```diff
@@ -179,7 +179,7 @@ def test_throws_on_negative(self):
 
 class TestSpotMaxPrice(unittest.TestCase):
     def test_valid_cases(self):
-        valid = [5, 5.12345, -1.0]
+        valid = [5, 5.12345, -1.0, 0.068, 0.071, 5.00000000]
```
Comment (qwordy) [inline src/aks-preview/azext_aks_preview/tests/latest/test_validators.py:182]: Is 5.00000000 valid? According to your new code, length('00000000') > 5.

**Comment by yungezz** [comment]
hi @arrownj could you pls help to review? thanks

File: src/aks-preview/azext_aks_preview/_validators.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (arrownj) [reply src/aks-preview/azext_aks_preview/_validators.py:217]: This is so tricky. Actually `isclose` function is not used correctly here, please try `isclose(namespace.spot_max_price * 100000 % 1, 0, abs_tol=1e-06)`.

File: src/aks-preview/azext_aks_preview/tests/latest/test_validators.py (hunk 2 above)
Comment (gtxistxgao) [reply src/aks-preview/azext_aks_preview/tests/latest/test_validators.py:182]: when it gets to the function, it becomes 5.0

File: src/aks-preview/azext_aks_preview/_validators.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (gtxistxgao) [reply src/aks-preview/azext_aks_preview/_validators.py:217]: which part is tricky? this line converts the float to a string "xxx.yyy" and check how many 'y' after '.'

[the remark this answers is not in the document]
Comment (gtxistxgao) [reply src/aks-preview/azext_aks_preview/_validators.py:217]: 0.071 * 100000 % 1 will be 0.9999999999990905
isclose(namespace.spot_max_price * 100000 % 1, 0, abs_tol=1e-06) will be false.
0.061 * 100000 % 1 will be 0.0
isclose(namespace.spot_max_price * 100000 % 1, 0, abs_tol=1e-06) will be true.

cannot use isclose to check how many decimal places for a float number

File: src/aks-preview/azext_aks_preview/tests/latest/test_validators.py (hunk 2 above)
Comment (gtxistxgao) [reply src/aks-preview/azext_aks_preview/tests/latest/test_validators.py:182]: in python, float number with all 0 after dot will be reduced to .0

Comment (qwordy) [reply src/aks-preview/azext_aks_preview/tests/latest/test_validators.py:182]: I see. It's a number, not a string.

File: src/aks-preview/azext_aks_preview/_validators.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (arrownj) [reply src/aks-preview/azext_aks_preview/_validators.py:217]: I'm wondering whether the value can be specified as `7.1e-2`. If yes, this solution may have problem to handle that. 

We can use `Decimal` to achieve it.
```suggestion
        if Decimal(str(namespace.spot_max_price)).as_tuple().exponent < -5:
```

If the value can only be xxx.yyy format, I'm ok with current solution.

**Review by arrownj — APPROVED** [review]

File: src/aks-preview/azext_aks_preview/_validators.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (gtxistxgao) [reply src/aks-preview/azext_aks_preview/_validators.py:217]: I am fine to block them if they input a 7.13e-2 as max price for a spot VM. :-p @jluk  FYI

**Pull request merged**